WMF Framework/Event Platform Client/EventPlatformClientLibrary.swift

```diff
@@ -0,0 +1,776 @@
+    }
+
+    /**
+     * Thread-safe synchronous buffering of an event
+     * - Parameter event: event to be buffered
+     */
+    func appendEventToInputBuffer(_ event: (stream: String, data: [String: NSCoding], domain: String?)) {
+        queue.sync {
+            /*
+             * Check if input buffer has reached maximum allowed size. Practically
+             * speaking, there should not have been over a hundred events
+             * generated when the user first launches the app and before the
+             * stream configuration has been downloaded and becomes available. In
+             * such a case we're just going to start clearing out the oldest
+             * events to make room for new ones.
+             */
+            if self.inputBuffer.count == self.inbutBufferLimit {
+                self.inputBuffer.remove(at: 0)
+            }
+            self.inputBuffer.append(event)
+        }
+    }
+
+    /**
+     * Thread-safe synchronous check if any events have been buffered
+     * - Returns: `true` if there are no buffered events, `false` if there are
+     */
+    func inputBufferIsEmpty() -> Bool {
+        queue.sync {
+            return self.inputBuffer.count == 0
+        }
+    }
+
+    /**
+     * Thread-safe synchronous removal of buffered event at index
+     * - Parameter index: The index from which to remove the buffered event in
+     *   the event buffer.
+     * - Returns: a previously buffered event
+     */
+    func removeInputBufferAtIndex(_ index: Int) -> (stream: String, data: [String: NSCoding], domain: String?)? {
+        queue.sync {
+            return self.inputBuffer.remove(at: index)
+        }
+    }
+
+    /**
+     * Thread-safe asynchronous caching of a stream's in-vs-out-of-sample
+     * determination
+     * - Parameter stream: name of stream to cache determination for
+     * - Parameter inSample: whether the stream was determined to be in-sample
+     *   this session
+     */
+    func cacheSamplingForStream(_ stream: String, inSample: Bool) {
+        queue.async {
+            self.samplingCache[stream] = inSample
+        }
+    }
+
+    /**
+     * Thread-safe synchronous retrieval of a stream's cached in-vs-out-of-sample determination
+     * - Parameter stream: name of stream to retrieve determination for from the cache
+     * - Returns: `true` if stream was determined to be in-sample this session, `false` otherwise
+     */
+    func getSamplingForStream(_ stream: String) -> Bool? {
+        queue.sync {
+            return self.samplingCache[stream]
+        }
+    }
+
+    /**
+     * Thread-safe asynchronous clearance of cached stream in-vs-out-of-sample determinations
+     */
+    func removeAllSamplingCache() {
+        queue.async {
+            self.samplingCache.removeAll()
+        }
+    }
+}
+
+//MARK: PeriodicWorker
+
+extension EPC: PeriodicWorker {
+    public func doPeriodicWork(_ completion: @escaping () -> Void) {
+        networkManager.httpTryPost()
```

Comment:
@tonisevener: Since EPC is extended to have `httpTryPost` which then calls network manager's `httpTryPost`, should this be `self.httpTryPost()` instead? Or should we remove

``` swift
/**
 * Passthrough method to tell `networkManager` to attempt to post its
 * scheduled events
 */
@objc public func httpTryPost() {
  networkManager.httpTryPost()
}
```

above?

Reply:
@bearloga Oh yeah, good catch. I can't remember why I thought we needed the passthrough method (maybe I didn't realize all calls would be able to flow through the `PeriodicWorker`), but it doesn't seem needed now. You can remove it.